### PR TITLE
gh-108826: Document `dis` module CLI and rename `_test` function to `main`

### LINKED ISSUE #108826: `dis` command line interface exists, but not documented
`dis` CLI exists de facto and is useful for disassembly of source code files or code from stdin, but it is not documented and mentioned as a "test program" in source code.

There was a previous discussion #88571 and PR #26714, but the last one is closed for unknown reason. I'll submit a new PR.

### Linked PRs
* gh-108827
* gh-110681
* gh-110689

### PR DESCRIPTION
**Pull request by chgnrdv**
Fixes #108826.
`dis` module has CLI, but it is not documented and declared as 'test program' in source code. This patch mentions CLI in documentation and renames the function where it is implemented.

----
:books: Documentation preview :books:: https://cpython-previews--108827.org.readthedocs.build/

* Issue: gh-108826

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2023-09-02-18-48-44.gh-issue-108826.KG7abS.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+:mod:`dis` module command line interface is now mentioned in documentation.
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2023-09-02-18-48-44.gh-issue-108826.KG7abS.rst]: I am not sure whether the Library section or the Documentation section are more suitable for this entry.

**Review by serhiy-storchaka — COMMENTED** [review]
Is it the right place for that section? I used to see the "Command-line interface" section at the end of the file, only followed by examples and howtos. But there are documents which do not follow this "custom", so I no longer sure.

**Review by serhiy-storchaka — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2023-09-02-18-48-44.gh-issue-108826.KG7abS.rst (hunk 1 above)
Comment (chgnrdv) [reply Misc/NEWS.d/next/Library/2023-09-02-18-48-44.gh-issue-108826.KG7abS.rst]: Different docs mention CLI in different sections of file. I was looking at `trace` and `pdb` docs when making these changes.
`dis` module has a preamble with brief introduction to its functionality, and I found the place below it suitable for CLI overview.

Yes, "Documentation" section seems more suitable for news entry, even if I made changes in library code. I'll submit a new one.

File: Doc/library/dis.rst [hunk 2]
```diff
@@ -63,6 +63,27 @@ the following command can be used to display the disassembly of
 
 (The "2" is a line number).
 
+Command-line interface
+----------------------
+
+The :mod:`dis` module can be invoked as a script from the command line.
+It is as simple as:
```
Comment (hugovk) [inline Doc/library/dis.rst]: Avoid saying things are "simple": https://www.knowledgeowl.com/blog/posts/dont-say-simply-jim-fisher/

```suggestion
The :mod:`dis` module can be invoked as a script from the command line:
```

File: Misc/NEWS.d/next/Documentation/2023-09-03-13-43-49.gh-issue-108826.KG7abS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+:mod:`dis` module command line interface is now mentioned in documentation.
```
Comment (hugovk) [inline Misc/NEWS.d/next/Documentation/2023-09-03-13-43-49.gh-issue-108826.KG7abS.rst]: Little nit:
```suggestion
:mod:`dis` module command-line interface is now mentioned in documentation.
```

**Review by hugovk — COMMENTED** [review]
Thanks for this, a couple of little suggestions :)

File: Doc/library/dis.rst (hunk 2 above)
Comment (chgnrdv) [reply Doc/library/dis.rst]: @hugovk I agree, but `trace`, `ast` and `tokenize` docs use this wording as well :)

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Please could you also update the anchor in https://docs.python.org/3.13/library/cmdline.html to link to the new CLI section of the `dis` page?

**Comment by chgnrdv** [comment]
#110249 added `--show-caches` command line option. I think it's OK if I will document it in this PR.

**Comment by hugovk** [comment]
Well spotted, thanks for adding it!

We'll have to make sure we don't include it in backports to 3.11 and 3.12. But we'll need a manual backport anyway, because we don't yet have `Doc/library/cmdline.rst` in those branches, and I also expect there will be some CLIs in 3.13 that don't belong in the other two.

**Comment by chgnrdv** [comment]
@hugovk, thanks for review and backport!

**Comment by hugovk** [comment]
Thanks for the PR!

**Pull request merged**